ydb/public/lib/ydb_cli/commands/interactive/yql_complete.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "c3_engine.h"
+#include "yql_syntax.h"
+
+#include <yql/essentials/parser/proto_ast/gen/v1_antlr4/SQLv1Antlr4Lexer.h>
+#include <yql/essentials/parser/proto_ast/gen/v1_antlr4/SQLv1Antlr4Parser.h>
+#include <yql/essentials/parser/proto_ast/gen/v1_ansi_antlr4/SQLv1Antlr4Lexer.h>
+#include <yql/essentials/parser/proto_ast/gen/v1_ansi_antlr4/SQLv1Antlr4Parser.h>
+
+#include <util/generic/string.h>
+#include <util/generic/vector.h>
+
+#include <string>
+
+namespace NYdb {
+    namespace NConsoleClient {
+
+        struct TCompletedToken final {
+            TStringBuf Content;
+            size_t SourcePosition;
+        };
+
+        enum class ECandidateKind {
+            Keyword,
+        };
+
+        // std::string is used to prevent copying into replxx api
+        struct TCandidate final {
+            ECandidateKind Kind;
+            std::string Content;
+
+            friend bool operator==(const TCandidate& lhs, const TCandidate& rhs) = default;
+        };
+
+        struct TCompletion final {
+            TCompletedToken CompletedToken;
+            TVector<TCandidate> Candidates;
+        };
+
+        class TYQLCompletionEngine final {
+            using TDefaultYQLGrammar = TAntlrGrammar<
+                NALPDefaultAntlr4::SQLv1Antlr4Lexer,
+                NALPDefaultAntlr4::SQLv1Antlr4Parser>;
+
+            using TAnsiYQLGrammar = TAntlrGrammar<
+                NALPAnsiAntlr4::SQLv1Antlr4Lexer,
+                NALPAnsiAntlr4::SQLv1Antlr4Parser>;
+
+        public:
+            TYQLCompletionEngine();
+
+            TCompletion Complete(TStringBuf prefix);
```

Comment:
Currently it works, but in future it should take full query and cursor position to be able to infer column types from the table schema found at the query tail. I left this as is, because it is hard to foresee everything in advance anyway.